### PR TITLE
Reduce search flicker.

### LINKED ISSUE #84499: Search tree blinks too much
Open search viewlet, search for "setPanelHidden" type character by character and notice how the tree blinks all the time fully.

The issue here is what you are doing to the tree, you should just call `refresh` on the tree and the id of the elements should be stable and there should not be so much blinking.

For an example of a tree that does not blink much even though the whole tree input is changed is the debug variables or the debug call stack tree. You can check it out as an example

fyi @joaomoreno @roblourens 

hello

### PR DESCRIPTION
**Pull request by JacksonKearl**
@roblourens made some changes... more complicated than I expected to deal with disposing past results, let me know what you think.

Closes #84499

### CODE REVIEW

File: src/vs/workbench/contrib/search/browser/searchView.ts [hunk 1]
```diff
@@ -1466,7 +1467,7 @@ export class SearchView extends ViewletPanel {
 
 			dom.append(messageEl, $('p', undefined, resultMsg));
 			this.reLayout();
-		} else if (!msgWasHidden) {
+		} else if (msgWasHidden) {
```
Comment (roblourens) [inline src/vs/workbench/contrib/search/browser/searchView.ts]: Is this right? I don't understand the purpose of the original code exactly

**Review by roblourens — APPROVED** [review]
Looks great! I noticed that if you have search results and click the "clear" button, then the message "found x results in y files" takes a second to disappear. Not sure whether that's from this change or not.

**Comment by JacksonKearl** [comment]
@roblourens good catch re. the message being slow to disappear, that's from this.

File: src/vs/workbench/contrib/search/browser/searchView.ts (hunk 1 above)
Comment (JacksonKearl) [reply src/vs/workbench/contrib/search/browser/searchView.ts]: Needed that for a bit, then just now went back and reverted it and the behavior is the same. 🤷‍♀️

**Pull request merged**